### PR TITLE
Maximumlikelihoodfactory enhancements

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

**Comment by jschueller** [comment]
@sofianehaddad, @regislebrun I'm not sure about d223794
also there are several places where we compute loglikelihoods and sometimes we return logminscalar, sometimes -maxscalar

File: python/test/t_MaximumLikelihoodFactory_std.py [hunk 1]
```diff
@@ -15,3 +15,45 @@
 inf_distribution = factory.build(sample)
 print(inf_distribution)
 
+class TruncatedFactory(object):
+    def __init__(self, distribution, lb=None, ub=None):
```
Comment (regislebrun) [inline python/test/t_MaximumLikelihoodFactory_std.py]: @jchueller any hope to have it in the C++ library? eg as the TruncatedDistributionFactory class?

Comment (jschueller) [inline python/test/t_MaximumLikelihoodFactory_std.py]: @regislebrun after more testing it seems optimizer have trouble getting the bounds rights, so I forget TruncatedDistributionFactory for now.

Comment (sofianehaddad) [inline python/test/t_MaximumLikelihoodFactory_std.py]: @jschueller why did you remove it? The problem is that we should set an inequality constraint lb < ub + constrains if parameters should be in [lb, ub]

**Comment by regislebrun** [comment]
+1 @jschueller @sofianehaddad we should return SpecFunc::LogMinNumericalScalar each time we compute the log of a zero PDF/likelihood, but I got invalid results in TrapezoidalFactory when I did it in MaximumLikelihoodFactory. Under time pressure, I prefered to leave the code unchanged just before the release.

File: python/test/t_MaximumLikelihoodFactory_std.py (hunk 1 above)
Comment (jschueller) [inline python/test/t_MaximumLikelihoodFactory_std.py]: @sofianehaddad yes, but there's no generic way of knowing which parameters should be in bounds
